src/frontends/sql/decode_tag_transform_operation_test.cc

```diff
@@ -70,22 +70,41 @@ TEST_P(DecodeTagTransformTest, DecodeTagTransformTest) {
 id: %u
 tag_transform : {
   transform_rule_name: "%s"
-  tag_precondition_ids: [ %s ]
+  tag_preconditions: [ %s ]
   transformed_node: { %s } })";
-  const auto &[id, rule_name, value_textproto_and_ids] = GetParam();
+  const auto &[id, rule_name, precondition_name_sequence,
+               value_textproto_and_ids] = GetParam();
   const auto &[value_textproto, value_ids] = value_textproto_and_ids;
 
+  absl::flat_hash_map<absl::string_view, uint64_t> precondition_name_to_id;
+  for (uint64_t i = 0; i < value_ids.size(); ++i) {
+    precondition_name_to_id.insert(
+        {precondition_name_sequence.at(i), value_ids.at(i)});
+  }
+
   Expression expr;
   EXPECT_TRUE(google::protobuf::TextFormat::ParseFromString(
       absl::StrFormat(kTagTransformTextprotoFormat, id, rule_name,
-                      absl::StrJoin(value_ids, ", "), value_textproto),
+                      absl::StrJoin(precondition_name_to_id, ", ",
+                                    [](std::string *out, auto pair) {
+                                      return absl::StrAppend(
+                                          out, absl::StrFormat(
+                                                   R"({ key: "%s" value: %d })",
+                                                   pair.first, pair.second));
+                                    }),
+                      value_textproto),
       &expr));
   Value decoded_value = DecodeExpression(expr, decoder_context);
 
-  std::vector<Value> expected_values =
-      MapIter<Value>(value_ids, [&decoder_context](uint64_t current_id) {
-        return decoder_context.GetValue(current_id);
-      });
+  std::vector<std::pair<std::string, Value>> expected_name_to_value_vec =
```

Comment:
auto `expected_name_to_value_vec` might be more readable?

Reply:
Certainly more concise, and (most of) the type is explicitly written on the RHS. Fixed.